### DISCUSSION TITLE
FreeTube v0.21.0 Beta

### DISCUSSION THREAD

**Discussion by PrestonN** [FreeTubeApp/FreeTube#5298] [Announcements]
## Summary
FreeTube v0.21.0 is out, and with it, our fastest minor release upgrade rollout in over two years. v0.19 came out after 10 months, v0.20 after 7 months, and v0.21 is here after just 3 months. With it, we have a myriad of features, user experience improvements, bug fixes, and performance improvements. 

Excluding dependency updates, Weblate translations, and nightly-only regression fixes, we have 110 (!) substantive merged PRs included in this release, and only 47 of them are linked to directly in this list. [Click here to see our full changelog.](https://github.com/FreeTubeApp/FreeTube/compare/v0.20.0-beta...v0.21.0-beta)

## Features
* Grid view for user playlists by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/4949
* Sort option for user playlists by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/4921
* Prevent adding duplicate videos into a playlist by default by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5044
* Add "Remove Duplicate Videos" button for user playlists by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5191
* Use more generic icon for `Quick bookmark` feature by @MarmadileManteater in https://github.com/FreeTubeApp/FreeTube/pull/4885
* Quick bookmark styling and behavioral improvements by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/5058
* Display `Subtitles` and `4k` tags on videos in search results when applicable by @dkshxd in https://github.com/FreeTubeApp/FreeTube/pull/5119
* Add <kbd>Ctrl</kbd>+<kbd>F</kbd> secondary search bar shortcut + trigger form submit on <kbd>Enter</kbd> binding by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/4970
* Show when Subscriptions / Trending / Most Popular were last updated by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/4380
* Allow filtering searches by features by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5125
* Play next/previous video after removing the current video from the playlist by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/5158
* Local API: Extract playlists on the auto-generated "Music" channel by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5250
* Local API: Extract releases on auto-generated artist topic channels by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5241
* Add Solarized Dark and Solarized Light color schemes by @DontBlameMe99 in https://github.com/FreeTubeApp/FreeTube/pull/5066
* Add option in Subscription Settings for a confirmation popup before unsubscribing by @msagr in https://github.com/FreeTubeApp/FreeTube/pull/4896
* Add settings section sorting option by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/5010
* Add more options to default playback rate menu by @msagr in https://github.com/FreeTubeApp/FreeTube/pull/4880
* Add icons to dropdowns by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/5006
* Improved spacing on the about page by @MarkusGaugg in https://github.com/FreeTubeApp/FreeTube/pull/5257
* Update About page to display correct Freetube logo based on currently set theme by @ducks in https://github.com/FreeTubeApp/FreeTube/pull/5126
* Mark destructive actions (e.g., deleting actions) with distinctive theming by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/4374
* Improve user playlist video search UX by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/4929
* Improve create & edit playlist UX by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5226
* Release notes formatting improvements from @absidue and @kommunarr

## Fixes
* Fix broken DeArrow thumbnails by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5259
* Fix incorrect published dates in the channel search with the Invidious API by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/4912
* Fix hide/show channel in ft-list-video by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5149
* Fix hide sharing actions also hiding hide/show channel by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5110
* Fix Channel live tab showing as empty for some channels by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/4914
* Fix window title displaying null while loading a video or channel by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/4913
* Fix user playlist reverse button unsetting itself by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/4947
* Fix prompts closing on button click in certain instances by @MarmadileManteater in https://github.com/FreeTubeApp/FreeTube/pull/5096
* Improve search filter styling & fix modal accessibility issues by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/3975
* Fix handling of emojis with ZWJ sequences in profile initials by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5023
* Fix External Player button not working on the Watch page by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5263
* Fix error when fetching deleted comment replies with the Local API by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5255
* Fix mouse backward and forward buttons not updating the in-app navigation buttons by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5242
* Fix incorrect Channel page sort values being displayed by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/5162
* Improve storyboard styling for tablet & mobile devices by @MarmadileManteater in https://github.com/FreeTubeApp/FreeTube/pull/4933
* Improve accessibility of playlist icon buttons by @sossost in https://github.com/FreeTubeApp/FreeTube/pull/4943

## Changes
* Replace Comment Auto Load setting with all-encompassing Auto Load Next Page setting by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/4565
* Add back matching author names appearing in video playlist search results by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/4919
* Increase default scrollbar size and make it larger on hover by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/5172
* Show error message when search is over the YouTube search limit of 100 characters by @dkshxd in https://github.com/FreeTubeApp/FreeTube/pull/4992
* Update Invidious instances list by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5288
* Use clearer SVG assets over blurrier PNGs by @HighlanderCZ in https://github.com/FreeTubeApp/FreeTube/pull/4898
* Only approve web API permission requests for permissions that FreeTube needs by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5022
* Bump electron from 29.1.6 to 31.0.1
* Enormous amount of performance improvements and build size reductions from @absidue
* Assorted code quality improvements by @absidue and @ChunkyProgrammer
* Changes to the About page styling by @efb4f5ff-1298-471a-8973-3d47447115dc

## New Contributors
* @HighlanderCZ with https://github.com/FreeTubeApp/FreeTube/pull/4898
* @msagr with https://github.com/FreeTubeApp/FreeTube/pull/4880
* @sossost with https://github.com/FreeTubeApp/FreeTube/pull/4943
* @dkshxd with https://github.com/FreeTubeApp/FreeTube/pull/4992
* @DontBlameMe99 with https://github.com/FreeTubeApp/FreeTube/pull/5066
* @MarkusGaugg with https://github.com/FreeTubeApp/FreeTube/pull/5257
* @ducks with https://github.com/FreeTubeApp/FreeTube/pull/5126

**Full Changelog**: https://github.com/FreeTubeApp/FreeTube/compare/v0.20.0-beta...v0.21.0-beta

<hr /><em>This discussion was created from the release <a href='https://github.com/FreeTubeApp/FreeTube/releases/tag/v0.21.0-beta'>FreeTube v0.21.0 Beta</a>.</em>

**Comment by hundehausen** [comment]
Awesome release! Any chance we can get build for Macs with Apple Silicon chips?

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
We cant give an ETA on that but you can look into https://github.com/FreeTubeApp/FreeTube/issues/2786#issuecomment-2130686731

**Reply by PikachuEXE** [reply]
I just woke up and the build need to be started manually
Try it now: https://github.com/PikachuEXE/homebrew-FreeTube
